### PR TITLE
Write logs to local, not config dir

### PR DESCRIPTION
**Pull request by franga2000**
As discussed in #3406

Tested:

- Platforms:
  - [x] Linux
  - [ ] Windows
  - [ ] Mac OS
 - Features:
   - [x] Dir structure and files created and written to as normal
   - [x] Old logs are gzipped correctly
   - [x] Creating a debug archive still includes log files

### CODE REVIEW

**Comment by FlexW** [comment]
Cool, thank you :) will you be able to test it on Windows and macOS aswell?

**Comment by franga2000** [comment]
Yes, but I'm still setting up toolchains on both. Should have results in a few hrs

File: src/libsync/configfile.cpp [hunk 1]
```diff
@@ -951,7 +951,7 @@ void ConfigFile::setAutomaticLogDir(bool enabled)
 
 QString ConfigFile::logDir() const
 {
-    const auto defaultLogDir = QString(configPath() + QStringLiteral("/logs"));
+    const auto defaultLogDir = QString(QStandardPaths::writableLocation(QStandardPaths::AppLocalDataLocation) + QStringLiteral("/logs"));
```
Comment (allexzander) [inline src/libsync/configfile.cpp:954]: @franga2000 I'd say, you need to change the corresponding part in `QString ConfigFile::configPath() const` if you want to only affect Linux. There is certain logic in `QString ConfigFile::configPath() const` that gets executed for Windows. Better try not to change the behavior.

**Review by allexzander — CHANGES_REQUESTED** [review]

File: src/libsync/configfile.cpp (hunk 1 above)
Comment (allexzander) [reply src/libsync/configfile.cpp:954]: @franga2000 Also, make sure **debug logs archive** can be generated via **Settings** dialog UI after this change.

**Comment by franga2000** [comment]
I haven't been able to build this to test on Windows or macOS as I only have VMs of them and everything takes ages to install. Any chance someone with a working build environment could give this a quick test (or at least compile and send it to me)?

**Comment by Rello** [comment]
Hello,

Thank you for your contribution to the Desktop Client with this pull request.
We are closing this request as it is outdated, no longer relevant (e.g., due to Qt 6), or does not align with the current roadmap.

We truly value your support and hope to see more contributions from you in the future!


Best regards,
Nextcloud Desktop Team

**Pull request closed without merging**